include/libopencm3/stm32/f4/dsi.h

```diff
@@ -0,0 +1,897 @@
+ */
+#define DSI_VPCR			MMIO32(DSI_BASE + 0x3CU)
+#define DSI_VPCR_VPSIZE_SHIFT		0
+#define DSI_VPCR_VPSIZE_MASK		0x3fff
+
+/*
+ * DSI Host Video Chunks Configuration Register
+ */
+#define DSI_VCCR			MMIO32(DSI_BASE + 0x40U)
+#define DSI_VCCR_NUMC_SHIFT		0
+#define DSI_VCCR_NUMC_MASK		0x1fff
+
+/*
+ * DSI Host Video Null Packet Configuration Register
+ */
+#define DSI_VNPCR			MMIO32(DSI_BASE + 0x44U)
+#define DSI_VNPCR_NPSIZE_SHIFT		0
+#define DSI_VNPCR_NPSIZE_MASK		0x1fff
+
+/*
+ * DSI Host Video HSA Configuration Register
+ */
+#define DSI_VHSACR			MMIO32(DSI_BASE + 0x48U)
+#define DSI_VHSACR_HSA_SHIFT		0
+#define DSI_VHSACR_HSA_MASK		0xfff
+
+/*
+ * DSI Host Video HBP Configuration Register
+ */
+#define DSI_VHBPCR			MMIO32(DSI_BASE + 0x4CU)
+#define DSI_VHBPCR_HBP_SHIFT		0
+#define DSI_VHBPCR_HBP_MASK		0xfff
+
+/*
+ * DSI Host Video Line Configuration Register
+ */
+#define DSI_VLCR			MMIO32(DSI_BASE + 0x50U)
+#define DSI_VLCR_HLINE_SHIFT		0
+#define DSI_VLCR_HLINE_MASK		0x7fff
+
+/*
+ * DSI Host Video VSA Configuration Register
+ */
+#define DSI_VVSACR			MMIO32(DSI_BASE + 0x54U)
+#define DSI_VVSACR_VSA_SHIFT		0
+#define DSI_VVSACR_VSA_MASK		0x3ff
+
+/*
+ * DSI Host Video VBP Configuration Register
+ */
+#define DSI_VVBPCR			MMIO32(DSI_BASE + 0x58U)
+#define DSI_VVBPCR_VBP_SHIFT		0
+#define DSI_VVBPCR_VBP_MASK		0x3ff
+
+/*
+ * DSI Host Video VFP Configuration Register
+ */
+#define DSI_VVFPCR			MMIO32(DSI_BASE + 0x5CU)
+#define DSI_VVFPCR_VFP_SHIFT		0
+#define DSI_VVFPCR_VFP_MASK		0x3ff
+
+/*
+ * DSI Host Video VA Configuration Register
+ */
+#define DSI_VVACR			MMIO32(DSI_BASE + 0x60U)
+#define DSI_VVACR_VA_SHIFT		0
+#define DSI_VVACR_VA_MASK		0x3fff
+
+/*
+ * DSI Host LTDC Command Configuration Register
+ */
+#define DSI_LCCR			MMIO32(DSI_BASE + 0x64U)
+#define DSI_LCCR_CMDSIZE_SHIFT		0
+#define DSI_LCCR_CMDSIZE_MASK		0xffff
+
+/*
+ * DSI Host Command mode Configuration Register
+ */
+#define DSI_CMCR			MMIO32(DSI_BASE + 0x68U)
+#define DSI_CMCR_MRDPS			(1 << 24)
+#define DSI_CMCR_DLWTX			(1 << 19)
+#define DSI_CMCR_DSR0TX			(1 << 18)
+#define DSI_CMCR_DSW1TX			(1 << 17)
+#define DSI_CMCR_DSW0TX			(1 << 16)
+/* Bit 15 reserved */
+#define DSI_CMCR_GLWTX			(1 << 14)
+#define DSI_CMCR_GSR2TX			(1 << 13)
+#define DSI_CMCR_GSR1TX			(1 << 12)
+#define DSI_CMCR_GSR0TX			(1 << 11)
+#define DSI_CMCR_GSW2TX			(1 << 10)
+#define DSI_CMCR_GSW1TX			(1 << 9)
+#define DSI_CMCR_GSW0TX			(1 << 8)
+/* Bits 7:2 Reserved */
+#define DSI_CMCR_ARE			(1 << 1)
+#define DSI_CMCR_TEARE			(1 << 0)
+
+/*
+ * DSI Host Generic Header Configuration Register
+ */
+#define DSI_GHCR			MMIO32(DSI_BASE + 0x6CU)
+#define DSI_GHCR_WCMSB_SHIFT		16
+#define DSI_GHCR_WCMSB_MASK		0xff
+#define DSI_GHCR_WCLSB_SHIFT		8
+#define DSI_GHCR_WCLSB_MASK		0xff
+#define DSI_GHCR_DATA1_SHIFT		16	/* data 1 in 'short' mode */
+#define DSI_GHCR_DATA1_MASK		0xff
+#define DSI_GHCR_DATA0_SHIFT		8	/* data 0 in 'short' mode */
+#define DSI_GHCR_DATA0_MASK		0xff
```

Comment:
Yeah, I figured that out, was just curious whether it made more sense to use the same defines, rather than repeating them, to hammer home that they are aliases for the same field. no biggy